lib/src/facts/aix/networking_resolver.cc

```diff
@@ -0,0 +1,231 @@
+#include <internal/facts/aix/networking_resolver.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <unordered_map>
+
+#include <inttypes.h>
+#include <net/if.h>
+#include <net/if_dl.h>
+#include <net/route.h>
+#include <netinet/in.h>
+#include <sys/ndd_var.h>
+#include <sys/kinfo.h>
+
+// This usage is recommended in several mailing lists, and is used at
+// least in Samba to query mac addresses. I saw some references to old
+// IBM publications also recommending its use, but couldn't actually
+// dig those pubs up.
+//
+// The structs it *returns* are in headers, though (specifically
+// mentioning usage of this function).
+//
+// This is all leftovers from ancient versions of the BSD networking
+// stack that every OS in the world has stolen/borrowed. It is pretty
+// gross in a lot of ways, but is what we have to work with on AIX.
+//
+// There does not appear to be a different or documented way to get
+// this information, outside of shelling out to tools which provide
+// human- rather than machine-readable output.
+extern "C" int getkerninfo(int, char*, int*, int32long64_t);
+
+namespace facter { namespace facts { namespace aix {
+    networking_resolver::data networking_resolver::collect_data(collection& facts) {
+        auto data = posix::networking_resolver::collect_data(facts);
+
+        // Query the kernel for the list of network interfaces and
+        // their associated addresses
+        data.interfaces = get_interfaces();
+
+        // query the network device descriptors from the kernel. This
+        // gives us physical information, such as mtu.
+        auto ndd_data = get_ndd_info();
+
+        for (auto& iface : data.interfaces) {
+            auto ndd_iter = ndd_data.find(iface.name);
+            if (ndd_iter != ndd_data.end()) {
+                iface.mtu = ndd_iter->second.ndd_mtu;
+            }
+        }
+
+        data.primary_interface = get_primary_interface();
+
+        return data;
+    }
+
+    networking_resolver::ndd_map networking_resolver::get_ndd_info() const {
+        int ksize = getkerninfo(KINFO_NDD, nullptr, nullptr, 0);
```

Comment:
This information could change between calls. Other places where we deal with this we've tended to loop several times in case the number of devices increased between calls. We should probably do that here.